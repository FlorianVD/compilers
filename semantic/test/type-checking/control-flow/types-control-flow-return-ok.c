// RUN-WITH-ARGS: -dump-type-table

int foo()
{
    return 2;
}

void bar()
{
    return;
}
