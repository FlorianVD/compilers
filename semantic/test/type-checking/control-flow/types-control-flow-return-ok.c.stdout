└── Program <7>
    ├── FuncDecl: returnType = 'int', name = 'foo' <3>
    │   └── CompoundStmt <2>
    │       └── ReturnStmt <1>
    │           └── IntLiteral: value = '2' <0>
    └── FuncDecl: returnType = 'void', name = 'bar' <6>
        └── CompoundStmt <5>
            └── ReturnStmt <4>
Type table:
0                   i64
